File: data/java/interfaces.java
interface Shape {
    double getArea();
}

class Rectangle implements Shape {
    double width;
    double height;

    Rectangle(double width, double height) {
        this.width = width;
        this.height = height;
    }

    @Override
    public double getArea() {
        return width * height;
    }
}

class Circle implements Shape {
    double radius;

    Circle(double radius) {
        this.radius = radius;
    }

    @Override
    public double getArea() {
        return Math.PI * radius * radius;
    }
}

public class Main {
    public static void main(String[] args) {
        Rectangle rectangle = new Rectangle(5, 3);
        Circle circle = new Circle(4);

        System.out.println("Rectangle Area: " + rectangle.getArea());  // Output: Rectangle Area: 15.0
        System.out.println("Circle Area: " + circle.getArea());  // Output: Circle Area: 50.26548245743669
    }
}

File: data/c#/interfaces.cs
using System;

interface IShape {
    double GetArea();
}

class Rectangle : IShape {
    public double Width { get; set; }
    public double Height { get; set; }

    public double GetArea() {
        return Width * Height;
    }
}

class Circle : IShape {
    public double Radius { get; set; }

    public double GetArea() {
        return Math.PI * Radius * Radius;
    }
}

class Program {
    static void Main(string[] args) {
        Rectangle rectangle = new Rectangle() { Width = 5, Height = 3 };
        Circle circle = new Circle() { Radius = 4 };

        Console.WriteLine("Rectangle Area: " + rectangle.GetArea());  // Output: Rectangle Area: 15
        Console.WriteLine("Circle Area: " + circle.GetArea());  // Output: Circle Area: 50.26548245743669
    }
}
